File: .taira-source-c80bf4d85a97/java/iroha_android/src/main/java/org/hyperledger/iroha/android/client/websocket/ToriiWebSocketListener.java
package org.hyperledger.iroha.android.client.websocket;

import java.nio.ByteBuffer;

/** Listener interface mirroring Torii WebSocket lifecycle events. */
public interface ToriiWebSocketListener {

  default void onOpen(final ToriiWebSocketSession session) {}

  default void onText(final ToriiWebSocketSession session, final CharSequence data, final boolean last) {}

  default void onBinary(final ToriiWebSocketSession session, final ByteBuffer data, final boolean last) {}

  default void onPing(final ToriiWebSocketSession session, final ByteBuffer message) {}

  default void onPong(final ToriiWebSocketSession session, final ByteBuffer message) {}

  default void onClose(final ToriiWebSocketSession session, final int statusCode, final String reason) {}

  default void onError(final ToriiWebSocketSession session, final Throwable error) {}
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/client/websocket/ToriiWebSocketListener.kt
package org.hyperledger.iroha.sdk.client.websocket

import java.nio.ByteBuffer

/** Listener interface mirroring Torii WebSocket lifecycle events. */
interface ToriiWebSocketListener {
    fun onOpen(session: ToriiWebSocketSession) {}
    fun onText(session: ToriiWebSocketSession, data: String) {}
    fun onBinary(session: ToriiWebSocketSession, data: ByteBuffer) {}
    fun onClose(session: ToriiWebSocketSession, statusCode: Int, reason: String) {}
    fun onError(session: ToriiWebSocketSession, error: Throwable) {}
}
